From 4c1ead4198aabde9ef33a0d3c7af27e10db8654b Mon Sep 17 00:00:00 2001
From: Vikram Bodireddy <vikram.bodireddy@intel.com>
Date: Thu, 1 Feb 2024 13:01:12 +0530
Subject: [PATCH] Add ApplyOptions D-bus property under Software

This change adds the ApplyOptions D-bus property
under xyz.openbmc_project.Software.BMC.Updater.
ApplyOptions is needed for BIOS NVRAM clear during
BIOS firmware update. ClearConfig attribute is passed
from RF to fwupd script.

Tested: Set and Get of ClearConfig from fwupd.sh works
        fine.

Signed-off-by: Vikram Bodireddy <vikram.bodireddy@intel.com>
Upstream-Status: Pending
---
 item_updater.hpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/item_updater.hpp b/item_updater.hpp
index 316113a..a0034d3 100644
--- a/item_updater.hpp
+++ b/item_updater.hpp
@@ -9,6 +9,7 @@
 #include <xyz/openbmc_project/Association/Definitions/server.hpp>
 #include <xyz/openbmc_project/Common/FactoryReset/server.hpp>
 #include <xyz/openbmc_project/Control/FieldMode/server.hpp>
+#include <xyz/openbmc_project/Software/ApplyOptions/server.hpp>
 
 #include <string>
 #include <vector>
@@ -24,7 +25,8 @@ using ItemUpdaterInherit = sdbusplus::server::object_t<
     sdbusplus::server::xyz::openbmc_project::common::FactoryReset,
     sdbusplus::server::xyz::openbmc_project::control::FieldMode,
     sdbusplus::server::xyz::openbmc_project::association::Definitions,
-    sdbusplus::server::xyz::openbmc_project::collection::DeleteAll>;
+    sdbusplus::server::xyz::openbmc_project::collection::DeleteAll,
+    sdbusplus::xyz::openbmc_project::Software::server::ApplyOptions>;
 
 namespace MatchRules = sdbusplus::bus::match::rules;
 using VersionClass = phosphor::software::manager::Version;
